backend/src/main/java/de/neuefische/CapStone/backend/service/CoffeeService.java

```diff
@@ -34,6 +34,14 @@ public CoffeeService(CoffeeRepository coffeeRepository, HubRepository hubReposit
         this.turnOffScheduleService = turnOffScheduleService;
     }
 
+    public CoffeeEntity findCoffeeByUid(String uid) {
+        Optional<CoffeeEntity> coffeeEntityOptional = coffeeRepository.findByDevice_Uid(uid);
+        if(coffeeEntityOptional.isEmpty()) {
+            throw new EntityNotFoundException("Device couldnt be found!");
```

Comment:
vielleicht noch die UID in die message hängen ?

Reply:
gute Idee 👍